client/src/partials/journal/journal.js

```diff
@@ -58,37 +63,74 @@ function JournalController(Transactions, Sorting, Grouping, Pagination, Filterin
+   *      other columns. This can be avoided by setting default sort and group.
+   *
+   *   2. To avoid some unmeaningfull column, we are using column such as project_name,
+   *      account_number, ... instead of project_id, account_id so the request to fetch data
+   *      should be a join to get expected columns name
+   *
+   *  @todo using ui-grid internazation if necessary for columns label
    */
-  vm.gridOptions.columnDefs = [
-    { field : 'trans_date', displayName : 'Date', cellFilter : 'date:"mediumDate"', filter : { condition : filtering.byDate } },
-    { field : 'description', displayName : 'Description' },
-    { field : 'account_number', displayName : 'Account' },
-    { field : 'debit_equiv', displayName : 'Debit' },
-    { field : 'credit_equiv', displayName : 'Credit' },
-    { field : 'trans_id',
-      displayName : 'Transaction',
-      sortingAlgorithm : sorting.transactionIds,
-      sort : { priority : 0, direction : 'asc' },
-      grouping : { groupPriority : 0 }
-    },
 
-    // @todo this should be formatted as a currency icon vs. an ID
-    { field : 'currency_id', displayName : 'Currency', visible: false },
+    columns = [
+      { field : 'uuid', displayName : 'ID'},
+      { field : 'project_name', displayName : 'Project'},
+      { field : 'period_summary', displayName : 'Period'},
+      { field : 'trans_date', displayName : 'Date', cellFilter : 'date:"mediumDate"', filter : { condition : filtering.byDate } },
+      { field : 'description', displayName : 'Description' },
+      { field : 'account_number', displayName : 'Account' },
+      { field : 'debit_equiv', displayName : 'Debit' },
+      { field : 'credit_equiv', displayName : 'Credit' },
+      { field : 'trans_id', 
+        displayName : 'Transaction', 
+        sortingAlgorithm : sorting.transactionIds,
+        sort : { priority : 0, direction : 'asc' },
+        grouping : { groupPriority : 0 }
+      },
+    
+      // @todo this should be formatted as a currency icon vs. an ID
+      { field : 'currency_id', displayName : 'Currency'},
+      
+      // @todo this should be formatted showing the debitor/credior
+      { field : 'entity_uuid', displayName : 'Recipient'}, 
+      { field : 'entity_type', displayName : 'Recipient Type'}, 
+
+      { field : 'reference_uuid', displayName : 'Reference Document'},
+      { field : 'record_uuid', displayName : 'Reference Document'},
+      { field : 'user', displayName : 'Responsible'},
+      
+      // @fixme this field should not come from the database as 'cc'
+      { field : 'cc_id', displayName : 'Cost Center'},
+      { field : 'pc_id', displayName : 'Profit Center'}
+    ];
+
+    vm.gridOptions.columnDefs = columns;
 
-    // @todo this should be formatted showing the debitor/credior
-    { field : 'deb_cred_uuid', displayName : 'Recipient', visible : false },
+    //adding additional service for columns config, the column service need columns tobe set before process
```

Comment:
I began to look for an appropriate event since yesterday but I did not find.
So there is no event fired when columns are available to the grid in my knowledge.